prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java

```diff
@@ -338,15 +337,15 @@ public APIResult submit(
         return new APIResult(APIResult.Status.SUCCEEDED, "Extension job submitted successfully:" + jobName);
     }
 
-    private Map<EntityType, List<Entity>> getEntityList(String extensionName, String jobName,
+    private TreeMap<EntityType, List<Entity>> getEntityList(String extensionName, String jobName,
                                                         List<FormDataBodyPart> feedForms,
                                                         List<FormDataBodyPart> processForms, InputStream config)
         throws FalconException, IOException{
         List<Entity> processes = getProcesses(processForms);
         List<Entity> feeds = getFeeds(feedForms);
         ExtensionType extensionType = getExtensionType(extensionName);
         List<Entity> entities;
-        Map<EntityType, List<Entity>> entityMap = new HashMap<>();
+        TreeMap<EntityType, List<Entity>> entityMap = new TreeMap<>();
```

Comment:
Believe you made this a tree map, so that during submission, feeds go first? But, while putting elements into this map, processes are going in first.